test/tests/functional/pbs_ralter.py

```diff
@@ -0,0 +1,1007 @@
+        msg += time.strftime(self.fmt, time.localtime(int(start)))
+        msg += " end="
+        msg += time.strftime(self.fmt, time.localtime(int(end)))
+        if (standing):
+            msg += " recurrence_rrule=" + rrule + " timezone="
+            msg += self.tzone
+
+        self.server.log_match(msg, regexp=True, interval=2, max_attempts=30)
+
+        if (ExpectSuccess):
+            attrs = {'reserve_state': (MATCH_RE, 'RESV_CONFIRMED|2')}
+            self.server.expect(RESV, attrs, id=rid)
+
+            msg = ".*Resv;" + rid + ";Reservation confirmed."
+            self.server.log_match(msg, regexp=True, interval=2,
+                                  max_attempts=30)
+
+            self.server.expect(RESV, attrs, id=rid)
+            acct_msg = ".*U;" + rid + ";requestor=" + TEST_USER.name + "@"
+            acct_msg += self.server.shortname
+
+            if (standing):
+                acct_msg += " recurrence_rrule=" + re.escape(rrule)
+                acct_msg += " timezone=" + re.escape(self.tzone)
+
+            self.server.accounting_match(acct_msg, regexp=True, interval=2,
+                                         max_attempts=30)
+        else:
+            msg = ".*Resv;" + rid + ";Reservation denied."
+            self.server.log_match(msg, regexp=True, interval=2,
+                                  max_attempts=30)
+
+        return rid, start, end
+
+    def check_resv_running(self, rid, offset=0, display=True):
+        """
+        Helper method to wait for reservation to start running.
+
+        param rid: Reservation id.
+        type  rid: string.
+
+        param offset: Time in seconds to wait for the reservation to start
+                      running.
+        type  offset: int.
+
+        param display: Whether to display a message or not, default - Yes.
+        type  display: bool
+        """
+        if (offset > 0):
+            if (display):
+                self.logger.info("Waiting for reservation to start running.")
+                time.sleep(offset)
+            else:
+                self.logger.info("Sleeping.")
+
+        attrs = {'reserve_state': (MATCH_RE, 'RESV_RUNNING|5')}
+        self.server.expect(RESV, attrs, id=rid)
+
+    def check_resv_finish(self, rid, duration, standing=False):
+        """
+        Helper method to wait for reservation to finish.
+
+        param rid: Reservation id.
+        type  rid: string.
+
+        param duration: Time in seconds to wait for the reservation to finish
+                      running.
+        type  duration: int.
+
+        param standing: Whether to display a message or not, default - Yes.
+        type  standing: bool
+        """
+        if (duration > 0):
+            self.logger.info("Waiting for reservation to finish.")
+        time.sleep(duration)
+        if (standing):
+            attrs = {'reserve_state': (MATCH_RE, 'RESV_CONFIRMED|2')}
+        else:
+            attrs = {'reserve_state': (MATCH_RE, 'RESV_FINISHED|6')}
```

Comment:
I saw that I am using this function only for standing reservations, so the code has changed and it doesn't have the race condition.